lib/libzpool/util.c

```diff
@@ -123,13 +123,13 @@ show_vdev_stats(const char *desc, const char *ctype, nvlist_t *nv, int indent)
 
 	for (c = 0; c < children; c++) {
 		nvlist_t *cnv = child[c];
-		char *cname, *tname;
+		char *cname = NULL, *tname;
 		uint64_t np;
 		if (nvlist_lookup_string(cnv, ZPOOL_CONFIG_PATH, &cname) &&
 		    nvlist_lookup_string(cnv, ZPOOL_CONFIG_TYPE, &cname))
 			cname = "<unknown>";
-		tname = calloc(1, strlen(cname) + 2);
-		(void) strcpy(tname, cname);
+		tname = umem_zalloc(strlen(cname) + 2, UMEM_NOFAIL);
+		(void) strlcpy(tname, cname, strlen(cname) + 2);
```

Comment:
Should you store strlen(cname) + 2 into a variable? A modern compiler might be able to optimize this out, but A) calling strlen() twice looks bad, and B) it'd more clearly link the semantics here if they shared a variable.